File: src/constants/URLs.js
import { APP_STAGES } from "./App"

// for Internet Explorer:
if (!window.location.origin) {
  window.location.origin = window.location.protocol + "//" + window.location.hostname + (window.location.port ? ':' + window.location.port: '');
}

const API_VERSION = 'v0';
const OFDB_API_LINK = (() => {
  switch (__STAGE__){
    case APP_STAGES.LOCAL:
      return "https://api.ofdb.io/";

    case APP_STAGES.BETA:
      return "https://api.ofdb.io/";

    case APP_STAGES.NIGHTLY:
      return "https://api.ofdb.io/";

    default:
      // production
      return "https://kartevonmorgen.org/api/";
  }
})() + API_VERSION;


const PUBLIC_RESOURSES = (() => {
  switch (__STAGE__) {
    case APP_STAGES.LOCAL:
      return "./public"
    case APP_STAGES.NIGHTLY:
      return "https://test.kartevonmorgen.org/public"
    case APP_STAGES.BETA:
      return "https://beta.kartevonmorgen.org/public"
    default:
      return "https://kartevonmorgen.org/public";
  }
})()

export const CC_LICENSE = {
    name: "creativecommons.org/publicdomain/zero/1.0/deed.de",
    link: "https://creativecommons.org/publicdomain/zero/1.0/deed.de"
  }

export const ODBL_LICENSE = {
    name: "opendatacommons.org/licenses/odbl/summary/",
    link: "https://opendatacommons.org/licenses/odbl/summary/"
  }

export const APP = {
    name: "kartevonmorgen.org",
    link: "https://kartevonmorgen.org"
  }

export const PROTOTYPE = {
    name: "mobile.kartevonmorgen.org",
    link: "https://mobile.kartevonmorgen.org"
  }

export const REPOSITORY = {
		name: "github.com/kartevonmorgen/",
    link: "https://github.com/kartevonmorgen/"
  }

export const FACEBOOK = {
    name: "facebook.com/vonmorgen",
    link: "https://www.facebook.com/vonmorgen"
  }

export const MAIL = {
    name: "info@kartevonmorgen.org",
    link: "mailto:info@kartevonmorgen.org"
  }

export const  OSM_ATTR = {
    name: "OpenStreetMap Contributors",
    link: "https://osm.org/copyright"
  }

export const TILE_SERVER = {
    link: "https://{s}.tile.openstreetmap.org/{z}/{x}/{y}.png"
  }

export const  TILE_SERVER_ATTR = {
    name: "Fix",
    link: "https://www.openstreetmap.org/fixthemap"
  }

export const OFDB_API = {
    link: OFDB_API_LINK
  }

export const TH_GEOCODER = {
    link: "https://geocoder.tilehosting.com/q/<query>.js?key=<key>"
  }

export const  NOMINATIM= {
    link: "https://nominatim.openstreetmap.org"
  }

export const  ROUTEPLANNERS = {

    default: {
      name: "Graphhopper Maps",
      link: "https://graphhopper.com/maps/?point=&point={addr}&locale=de-DE&vehicle=bike&weighting=fastest&elevation=false&use_miles=false&layer=Omniscale"
    },
    apple: {
      name: "Apple Maps",
      link: "http://maps.apple.com/?daddr={addr}&dirflg=w"
    },
    android: {
      name: "GeoURI",
      link: "geo:{lat},{lng}"
    }
  }

export const DONATE = {
    name: "www.betterplace.org/de/projects/36213",
    link: "https://www.betterplace.org/de/projects/36213"
  }

export const JOB_ADS = {
    name: "www.betterplace.org/de/organisations/vonmorgen",
    link: "https://www.betterplace.org/de/organisations/vonmorgen"
  }

export const ENTRY_HISTORY = {
    name: "EntryOpenFairDB",
    link: "https://openfairdb.org/entries",
  }

export const  EVENT_HISTORY = {
    name: "EventOpenFairDB",
    link: "https://openfairdb.org/events"
  }

export const  PROMINENT_TAGS = {
    link: "https://blog.vonmorgen.org/prominent-tags/"
  }

export const  CORS_PROXY = {
    // this is used to bypass cors, one of it's use cases is to fetch prominent tags
    link: "https://cors-anywhere.herokuapp.com"
  }

export const  PUBLIC_RESOURCES = {
    link: PUBLIC_RESOURSES
}

File: src/constants/URLs.rs
use crate::constants::App;
pub const API_VERSION: &str = "v0";
pub fn OFDB_API_LINK(stage: &App::APP_STAGES) -> String {
    let url = match stage {
        App::APP_STAGES::LOCAL => "/api/",
        App::APP_STAGES::NIGHTLY => "https://nightly.ofdb.io/",
        App::APP_STAGES::PRODUCTION => "https://api.ofdb.io/",
    };
    format!("{}{}", url, API_VERSION)
}
pub mod CC_LICENSE {
    pub const name: &str = "creativecommons.org/publicdomain/zero/1.0/deed.de";
    pub const link: &str = "https://creativecommons.org/publicdomain/zero/1.0/deed.de";
}
pub mod ODBL_LICENSE {
    pub const name: &str = "opendatacommons.org/licenses/odbl/summary/";
    pub const link: &str = "https://opendatacommons.org/licenses/odbl/summary/";
}
pub mod APP {
    pub const name: &str = "kartevonmorgen.org";
    pub const link: &str = "https://kartevonmorgen.org";
}
pub mod PROTOTYPE {
    pub const name: &str = "prototyp.kartevonmorgen.org";
    pub const link: &str = "https://prototyp.kartevonmorgen.org";
}
pub mod REPOSITORY {
    pub const name: &str = "github.com/flosse/kartevonmorgen";
    pub const link: &str = "https://github.com/flosse/kartevonmorgen";
}
pub mod FACEBOOK {
    pub const name: &str = "facebook.com/vonmorgen";
    pub const link: &str = "https://www.facebook.com/vonmorgen";
}
pub mod MAIL {
    pub const name: &str = "info@kartevonmorgen.org";
    pub const link: &str = "mailto:info@kartevonmorgen.org";
}
pub mod OSM_ATTR {
    pub const name: &str = "OpenStreetMap";
    pub const link: &str = "https://osm.org/copyright";
}
pub mod TILE_SERVER {
    pub const link: &str = "https://maps.wikimedia.org/osm-intl/{z}/{x}/{y}{r}.png";
}
pub mod TILE_SERVER_ATTR {
    pub const name: &str = "Wikimedia";
    pub const link: &str = "https://wikimediafoundation.org/wiki/Maps_Terms_of_Use";
}
pub use self::OFDB_API_LINK as OFDB_API;
pub mod TH_GEOCODER {
    pub const link: &str = "https://geocoder.tilehosting.com/q/<query>.js?key=<key>";
}
pub mod NOMINATIM {
    pub const link: &str = "https://nominatim.openstreetmap.org";
}
pub mod ROUTEPLANNERS {
    pub mod default {
        pub const name: &str = "Graphhopper Maps";
        pub const link : &str = "https://graphhopper.com/maps/?point=&point={addr}&locale=de-DE&vehicle=bike&weighting=fastest&elevation=false&use_miles=false&layer=Omniscale";
    }
    pub mod apple {
        pub const name: &str = "Apple Maps";
        pub const link: &str = "http://maps.apple.com/?daddr={addr}&dirflg=w";
    }
    pub mod android {
        pub const name: &str = "GeoURI";
        pub const link: &str = "geo:{lat},{lng}";
    }
}
pub mod DONATE {
    pub const name: &str = "www.betterplace.org/de/projects/36213";
    pub const link: &str = "https://www.betterplace.org/de/projects/36213";
}
pub mod JOB_ADS {
    pub const name: &str = "www.betterplace.org/de/organisations/vonmorgen";
    pub const link: &str = "https://www.betterplace.org/de/organisations/vonmorgen";
}
